### PR TITLE
Black formatter setup

### PR DESCRIPTION
**Pull request by Didah**
Configuration for Black + formatting of the whole project using that configuration.

### CODE REVIEW

File: software/contrib/polyrhythmic_sequencer.py
Comment (awonak) [inline software/contrib/polyrhythmic_sequencer.py:58]: Is it possible to configure Black to ignore certain formatting rules, or allow us to add a comment to ignore and not format the preceding line or block? This formatting change doesn't add value in my opinion, and removes context the previous formatting gave.

Comment (Didah) [reply software/contrib/polyrhythmic_sequencer.py:58]: Yes! We could use `# fmt: off` and `# fmt: on` to ignore blocks. I'll go ahead and revert this instance already.

File: .git-blame-ignore-revs
Comment (mjaskula) [inline .git-blame-ignore-revs:8]: [question] I've never used this feature, I'm wondering if we would want to wait until this PR is ready to merge, squash it into a single commit, and then add _that_ commit to this file. This would be to be able to ignore this entire effort at once.

Comment (mjaskula) [inline .git-blame-ignore-revs:8]: [question] I've never used this feature, I'm wondering if we would want to wait until this PR is ready to merge, squash it into a single commit, and then add _that_ commit to this file. This would be to be able to ignore this entire effort at once.

File: .github/workflows/continuous_integration.yml
Comment (mjaskula) [inline .github/workflows/continuous_integration.yml:24]: [question] my assumption is that this will cause a failure if any files aren't properly formatted. Is that correct?

Comment (mjaskula) [inline .github/workflows/continuous_integration.yml:24]: [question] my assumption is that this will cause a failure if any files aren't properly formatted. Is that correct?

**Review by mjaskula — APPROVED** [review]
This is great, thanks for taking the time to do this. I've added a few questions and discussion points. I don't think that we should be formatting the contrib directory, but let's have a discussion first before you reverse that part.

In addition, if we are adding this requirement, we should update the contributing.md file to include it. Let me know if you'd like to tackle that documentation or not.

File: software/contrib/bernoulli_gates.py
Comment (Didah) [reply software/contrib/bernoulli_gates.py:15]: I see what you're saying, agreed!

File: .git-blame-ignore-revs
Comment (Didah) [reply .git-blame-ignore-revs:8]: Good point, will move this out of the PR so we can add it after the eventual merge.

File: software/contrib/bernoulli_gates.py
Comment (Didah) [reply software/contrib/bernoulli_gates.py]: Added contrib directories `software/contrib` and `tests/contrib` to config so they are ignored by Black

File: .git-blame-ignore-revs
Comment (Didah) [reply .git-blame-ignore-revs]: Removed this - we can simply add the final commit hash after merging the PR indeed

File: contributing.md
Comment (awonak) [inline contributing.md:191]: [required] Can you also add a line to show folks how they can optionally run black as a one-off on their own contrib? Something like the following:

Optionally, you can also manually run `black` as a one-off to format your contrib script:

```
$ black software/contrib/polyrhythmic_sequencer.py
```
To prevent `black` from reformating certain parts of your code (which should be rare), start a block with `# fmt: off` and end with `# fmt: on`, or end a line with `# fmt: skip`.

**Review by awonak — APPROVED** [review]

File: contributing.md
Comment (Didah) [reply contributing.md]: Great suggestion! Added

**Comment by Didah** [comment]
This closes issue #144

**Pull request merged**
